### PR TITLE
refactor: follow style guide

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

File: src/nvim/decoration.c [hunk 1]
```diff
@@ -826,7 +826,7 @@ static void buf_signcols_validate_range(buf_T *buf, int row1, int row2, int add)
   int *overlap = xcalloc(sizeof(int), (size_t)(row2 + 1 - row1));
 
   // First find the number of overlapping signs at "row1".
-  (void)marktree_itr_get_overlap(buf->b_marktree, currow, 0, itr);
+  marktree_itr_get_overlap(buf->b_marktree, currow, 0, itr);
```
Comment (luukvbaal) [inline src/nvim/decoration.c:829]: Removing this (and others) may trigger a coverity warning. This was [added](https://github.com/neovim/neovim/pull/26518) recently to suppress one at least.

Comment (dundargoc) [reply src/nvim/decoration.c:829]: We should disable this dumb warning then IMO rather than try to appease coverity.

**Comment by justinmk** [comment]
This PR looks related to these new coverity warnings:

```
New defect(s) Reported-by: Coverity Scan
Showing 4 of 4 defect(s)


** CID 478868:  Insecure data handling  (TAINTED_SCALAR)


________________________________________________________________________________________________________
*** CID 478868:  Insecure data handling  (TAINTED_SCALAR)
/src/nvim/tag.c: 2123 in findtags_get_all_tags()
2117           break;
2118         }
2119     
2120     line_read_in:
2121     
2122         if (st->vimconv.vc_type != CONV_NONE) {
>>>     CID 478868:  Insecure data handling  (TAINTED_SCALAR)
>>>     Passing tainted expression "*st->lbuf" to "findtags_string_convert", which uses it as an offset.
2123           findtags_string_convert(st);
2124         }
2125     
2126         // When still at the start of the file, check for Emacs tags file
2127         // format, and for "not sorted" flag.
2128         if (st->state == TS_START) {

** CID 478867:  Insecure data handling  (TAINTED_SCALAR)


________________________________________________________________________________________________________
*** CID 478867:  Insecure data handling  (TAINTED_SCALAR)
/src/nvim/spellfile.c: 2056 in spell_read_aff()
2050           continue;
2051         }
2052     
2053         // Convert from "SET" to 'encoding' when needed.
2054         xfree(pc);
2055         if (spin->si_conv.vc_type != CONV_NONE) {
>>>     CID 478867:  Insecure data handling  (TAINTED_SCALAR)
>>>     Passing tainted expression "rline" to "string_convert", which uses it as an offset.
2056           pc = string_convert(&spin->si_conv, rline, NULL);
2057           if (pc == NULL) {
2058             smsg(0, _("Conversion failure for word in %s line %d: %s"),
2059                  fname, lnum, rline);
2060             continue;
2061           }

** CID 478866:    (TAINTED_SCALAR)


________________________________________________________________________________________________________
*** CID 478866:    (TAINTED_SCALAR)
/src/nvim/memline.c: 1556 in swapfile_info()
1550     
1551             msg_puts(_("\n          modified: "));
1552             msg_puts(b0.b0_dirty ? _("YES") : _("no"));
1553     
1554             if (*(b0.b0_uname) != NUL) {
1555               msg_puts(_("\n         user name: "));
>>>     CID 478866:    (TAINTED_SCALAR)
>>>     Passing tainted expression "b0.b0_uname" to "msg_outtrans", which uses it as an offset.
1556               msg_outtrans(b0.b0_uname, 0);
1557             }
1558     
1559             if (*(b0.b0_hname) != NUL) {
1560               if (*(b0.b0_uname) != NUL) {
1561                 msg_puts(_("   host name: "));
/src/nvim/memline.c: 1565 in swapfile_info()
1559             if (*(b0.b0_hname) != NUL) {
1560               if (*(b0.b0_uname) != NUL) {
1561                 msg_puts(_("   host name: "));
1562               } else {
1563                 msg_puts(_("\n         host name: "));
1564               }
>>>     CID 478866:    (TAINTED_SCALAR)
>>>     Passing tainted expression "b0.b0_hname" to "msg_outtrans", which uses it as an offset.
1565               msg_outtrans(b0.b0_hname, 0);
1566             }
1567     
1568             if (char_to_long(b0.b0_pid) != 0) {
1569               msg_puts(_("\n        process ID: "));
1570               msg_outnum((int)char_to_long(b0.b0_pid));
/src/nvim/memline.c: 1548 in swapfile_info()
1542             msg_puts(_("         [garbled strings (not nul terminated)]"));
1543           } else {
1544             msg_puts(_("         file name: "));
1545             if (b0.b0_fname[0] == NUL) {
1546               msg_puts(_("[No Name]"));
1547             } else {
>>>     CID 478866:    (TAINTED_SCALAR)
>>>     Passing tainted expression "b0.b0_fname" to "msg_outtrans", which uses it as an offset.
1548               msg_outtrans(b0.b0_fname, 0);
1549             }
1550     
1551             msg_puts(_("\n          modified: "));
1552             msg_puts(b0.b0_dirty ? _("YES") : _("no"));
1553     

** CID 478865:  Insecure data handling  (TAINTED_SCALAR)


________________________________________________________________________________________________________
*** CID 478865:  Insecure data handling  (TAINTED_SCALAR)
/src/nvim/memline.c: 3368 in findswapname()
3362                 process_running = swapfile_process_running(&b0, fname);
3363     
3364                 // If the swapfile has the same directory as the
3365                 // buffer don't compare the directory names, they can
3366                 // have a different mountpoint.
3367                 if (b0.b0_flags & B0_SAME_DIR) {
>>>     CID 478865:  Insecure data handling  (TAINTED_SCALAR)
>>>     Passing tainted expression "b0.b0_fname" to "path_tail", which uses it as an offset.
3368                   if (path_fnamecmp(path_tail(buf->b_ffname),
3369                                     path_tail(b0.b0_fname)) != 0
3370                       || !same_directory(fname, buf->b_ffname)) {
3371                     // Symlinks may point to the same file even
3372                     // when the name differs, need to check the
3373                     // inode too.
```

**Comment by dundargoc** [comment]
How are they related if the warnings first started showing up today?

**Pull request merged**